src/parlant/core/engines/alpha/engine.py

```diff
@@ -139,6 +139,48 @@ async def process(
             self._logger.critical(f"Critical processing error: {traceback.format_exception(exc)}")
             raise
 
+    @override
+    async def utter(
+        self,
+        context: Context,
+        event_emitter: EventEmitter,
+        actions: Sequence[UtteranceRequest],
+    ) -> bool:
+        interaction_state = await self._load_interaction_state(context)
+        session_id = context.session_id
+
+        try:
+            with self._logger.operation(f"Uttering actions '{actions}' for session {session_id}"):
+                await self._do_utter(context, interaction_state, event_emitter, actions)
+            return True
+
+        except asyncio.CancelledError:
+            self._logger.warning(f"Uttering actions for session {session_id} was cancelled.")
+            return False
+
+        except Exception as exc:
+            formatted_exception = traceback.format_exception(type(exc), exc, exc.__traceback__)
+            self._logger.error(
+                f"Error during uttering actions for session {session_id}: {formatted_exception}"
```

Comment:
"Error during uttering for session..." not "Error during uttering actions for session..."